### ISSUE TITLE
Manual join on aggregates are not supported

### ISSUE THREAD

**Issue by Shaddix** [ethanli83/EFSqlTranslator#38]
Let's say I want to display the list of Blogs and for each blog I want to display the Post which has maximum of likes (in reality, quite often the last post is displayed, i.e. having Max Date).

So, I want a query like that:
```
var query = db.Posts.GroupBy(x => x.BlogId).Select(x => new
                {
                    BlogId = x.Key,
                    MaxLikes = x.Max(z => z.LikeCount)
                });

                var query1 = db.Posts.
                    Join(
                        query,
                        (p, b) => p.BlogId == b.BlogId && p.LikeCount == b.MaxLikes,
                        (p, b) => new { PId = p.PostId, b.BlogId },
                        DbJoinType.LeftInner);
```

fix is in the works

**Comment by Shaddix** [comment]
@ethanli83 could you please help me with this one?
I added a couple of tests to 
https://github.com/Shaddix/EFSqlTranslator/blob/ManualJoinOnAggregation/EFSqlTranslator.Tests/TranslatorTests/ManualJoinTranslationTests.cs

and fixed an exception happening when join on aggregate. 
I even make these new test pass by commenting lines 45-47 in JoinTranslator.cs, but that's obviously wrong (and it makes the other Manual join test fail).

Could you please help me with this issue?

**Comment by ethanli83** [comment]
Of course @Shaddix, I will have a look at it.

**Comment by ethanli83** [comment]
hi @Shaddix,

I have fixed the issue and pushed out a new version 1.2.9. It also contains your pull requests.

**Comment by Shaddix** [comment]
thanks a lot!
